src/crimson/osd/main.cc

```diff
@@ -263,8 +263,8 @@ int main(int argc, char* argv[])
               local_conf().get_val<uuid_d>("osd_uuid"),
               local_conf().get_val<uuid_d>("fsid")).get();
           }
-          seastar::engine().at_exit([&] {
-            return osd.stop();
+          auto stop_osd = seastar::defer([&] {
+            osd.stop().get();
```

Comment:
nit: type in the commit msg. Suggested:

```
since we do the startup in a seastar thread, we have the luxury of doing
cleanup using the RAII machinery.
```

Reply:
fixed.